### PR TITLE
[2.0] Bump to php 7.1

### PR DESCRIPTION
**Pull request by Jean85**
This is to start the work planned in #670. In this branch I'll plan to just clean up everything related to PHP <7.1, update dependencies and make the build green.

~~This is still work in progress!~~ Done, everything is fine!

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,11 @@
...
 
 php:
-  - 5.6
-  - 7.0
   - 7.1
   - 7.2
 
 matrix:
   fast_finish: true
-  exclude:
```
[the remark this answers is not in the document]
Comment (Jean85) [reply .travis.yml]: We cannot do it while having PHP-CS-Fixer as a dependency, see [this failed job](https://travis-ci.org/getsentry/sentry-php/jobs/440994257#L486).

The only way that I can think of is removing it on the fly before the job. I'll try.

**Comment by Jean85** [comment]
I've disabled the build on Scrutinizer, it was just a duplication of what we already do on Travis.

I've fixed the nightly build, but it's failing for a stupid reason: https://travis-ci.org/getsentry/sentry-php/jobs/441521282#L622-L623

What should we do?

**Comment by Jean85** [comment]
I've fixed the "stupid reason". It's now all green! Should I merge?

**Comment by ste93cry** [comment]
Since you worked on the `scrutinizer.yml` file, can we also enable the new [PHP Analysis](https://scrutinizer-ci.com/docs/tools/php/php-analyzer/guides/migrate_to_new_php_analysis)?

**Comment by Jean85** [comment]
After a few back and forth with @ste93cry, we addressed all that needed to be fixed. Only one pending is the fact that Scrutinizer still insists in running CS checks, but it's not in sync with our rules. We can only ignore that for now.

**Pull request merged**